### ISSUE TITLE
DataGrid class needs to be rewritten to support so much records

### ISSUE THREAD

**Issue by helgispbru** [evolution-cms/evolution#983]
I have 1.4.8 with 500 thousand records in _event_log.

When I open System Events the result page is empty.

In php error log I see:
[23-Jan-2019 18:49:09 Europe/Moscow] PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 48192384 bytes) in 
 .../public_html/manager/includes/extenders/dbapi.mysqli.class.inc.php on line 157

That looks like it tries to get all records in one request.

I looked at /manager/actions/eventlog.dynamic.php and there is only 1 sql select and it is without any LIMIT command. So it really get all records in one request. o_O

As I see pagination and everything about it is made by DataGrid class, so I thing eventlog.dynamic.php and DataGrid needs to be rewritten to support so much records.

This can be done, for example, with SQL_CALC_FOUND_ROW which does not request all records, but count quantity of records with current WHERE in request.

**Comment by AgelxNash** [comment]
In version 2.x we must remove the class DataGrid.

**Comment by AgelxNash** [comment]
The DataGrid class gets a query that has already been executed. Rewriting can cause big backward compatibility issues.

**Comment by helgispbru** [comment]
I agree, that DataGrid class should only show that it get. If you think to remove DataGrid from 2.x it has no reason to rewrite it now.

In my situation I just add query limit to "0,1500" to /manager/actions/eventlog.dynamic.php and now it's enough.

I think you can close this issue. Thanks!
